src/main/java/seedu/address/model/person/Food.java

```diff
@@ -61,6 +77,35 @@ public boolean isSamePerson(Food otherFood) {
                 && otherFood.getName().equals(getName());
     }
 
+    private boolean hasTag(String tagName) {
+        return tags.stream().anyMatch(tag -> tag.tagName.equals(tagName));
+    }
+
+    /**
+     * Returns a String representing the earliest meal tag on this {@code Food}.
+     * @return "B" for breakfast, "L" for lunch, "D" for dinner and "X" for no tag.
+     */
+    public String getEarliestMealTag() {
+        if (hasTag("breakfast")) {
+            return "B";
+        } else if (hasTag("lunch")) {
+            return "L";
+        } else if (hasTag("dinner")) {
+            return "D";
+        } else {
+            return "X";
+        }
```

Comment:
Instead of returning a magic literal as described [here](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers), maybe it's better to define it as a constant first, for e.g. `private static final String BREAKFAST = "B"`? 🤔